cluster/nodes.go

```diff
@@ -0,0 +1,230 @@
+			  VALUES
+				($1, $2, $3, $4)`
+	tx, err := ctx.MainTx()
+	if err != nil {
+		return nil, err
+	}
+	// Execute query
+	_, err = tx.Exec(query, node.ID, node.Name, node.K8sLabel, ctx.WhoAmI)
+	if err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+
+func GetNodeByName(ctx *Context, name string) (*Node, error) {
+	// Get user from tenants database
+	query := `
+		SELECT 
+				n.id,n.name,n.k8s_label
+			FROM 
+				nodes n
+			WHERE n.name=$1 LIMIT 1`
+
+	row := GetInstance().Db.QueryRow(query, name)
+	node := Node{}
+	err := row.Scan(&node.ID, &node.Name, &node.K8sLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	return &node, nil
+}
+
+type NodeVolume struct {
+	ID        uuid.UUID
+	NodeID    *uuid.UUID
+	MountPath string
+	Num       int32
+}
+
+func NewVolume(nodeID *uuid.UUID, mountPath string) (*NodeVolume, error) {
+	var re = regexp.MustCompile(`^(\/[a-zA-Z0-9_-]+)+$`)
+	if !re.MatchString(mountPath) {
+		errMsg := fmt.Sprintf("invalid mount path: `%s`", mountPath)
+		return nil, errors.New(errMsg)
+	}
+	return &NodeVolume{ID: uuid.NewV4(), NodeID: nodeID, MountPath: mountPath}, nil
+}
+
+// VolumeAdd adds a new volume to a node
+func VolumeAdd(ctx *Context, nodeID *uuid.UUID, mountPoint string) (*NodeVolume, error) {
+	// if we can instantiate a new volume from the data, insert it
+	volume, err := NewVolume(nodeID, mountPoint)
+	if err != nil {
+		return nil, err
+	}
+	query := `INSERT INTO
+				node_volumes ("num", "id", "node_id", "mount_path", "sys_created_by")
+				SELECT COUNT(*)+1, $2, $3, $4, $5 FROM node_volumes WHERE node_id=$1`
+
+	tx, err := ctx.MainTx()
+	if err != nil {
+		return nil, err
+	}
+	// Execute query
+	_, err = tx.Exec(query, volume.NodeID, volume.ID, volume.NodeID, volume.MountPath, ctx.WhoAmI)
+	if err != nil {
+		return nil, err
+	}
+	return volume, nil
+}
+
+// Creates a storage cluster in the DB
+func AssignNodeToStorageCluster(ctx *Context, nodeID *uuid.UUID, storageClusterID *uuid.UUID) error {
+	// TODO: Validate the symmetry of disk of this node to other existing nodes on the storage cluster
+	tx, err := ctx.MainTx()
+	if err != nil {
+		return err
+	}
+	// Create association m-n
+	query :=
+		`INSERT INTO
+				storage_cluster_nodes ("num","storage_cluster_id", "node_id", "sys_created_by")
+			  SELECT COUNT(*)+1 ,$2, $3 ,$4 FROM storage_cluster_nodes WHERE storage_cluster_id=$1`
+
+	if _, err = tx.Exec(query, storageClusterID, storageClusterID, nodeID, ctx.WhoAmI); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Returns a list of nodes for a storage group
+func GetNodesForStorageGroup(ctx *Context, storageGroupID *uuid.UUID) ([]*StorageGroupNode, error) {
+	queryNodes := `
+		SELECT 
+				n.id,n.name,n.k8s_label,scn.num
+		FROM 
+			nodes n
+		LEFT JOIN storage_cluster_nodes scn ON n.id = scn.node_id
+		LEFT JOIN storage_groups sg ON scn.storage_cluster_id = sg.storage_cluster_id
+		WHERE 
+		      sg.id = $1`
+
+	rows, err := GetInstance().Db.Query(queryNodes, storageGroupID)
```

Comment:
This needs to be like this else it won't get the storage groups created in the earlier steps since they haven't been committed.
```suggestion
	rows, err := ctx.mainTx.Query(queryNodes, storageGroupID)
```